front/migrations/20240322_content_fragments_refactor.ts

```diff
@@ -0,0 +1,150 @@
+import { Op } from "sequelize";
+
+import { Conversation, Message, Workspace } from "@app/lib/models";
+import {
+  ContentFragmentResource,
+  storeContentFragmentText,
+} from "@app/lib/resources/content_fragment_resource";
+import { ContentFragmentModel } from "@app/lib/resources/storage/models/content_fragment";
+
+const { LIVE } = process.env;
+
+async function migrateContentFragment(
+  cfMessage: Message,
+  workspaceId: string,
+  conversationId: string
+) {
+  const cf = ContentFragmentResource.fromMessage(cfMessage);
+  // if textUrl is null, upload content to GCS and set textUrl to the uploaded file
+  // value (also set textBytes to the number of bytes in the content)
+  // @ts-expect-error textUrl was removed from ContentFragmentResource
+  if (!cf.textUrl) {
+    const fileUrl = await storeContentFragmentText({
+      workspaceId,
+      conversationId,
+      messageId: cfMessage.sId,
+      // @ts-expect-error content was removed from ContentFragmentResource
+      content: cf.content,
+    });
+    await cf.update({
+      // @ts-expect-error textUrl was removed from ContentFragmentResource
+      textUrl: fileUrl,
```

Comment:
I'm probably missing something but we remove `textUrl` from the model and in this migration we try to set it 🤔.

Reply:
This migration was indeed done on 22 march => at the time we thought we would keep textUrl
Yesterday, we decided it was not useful => so we delete